### ISSUE TITLE
[Bug]: Could not connect to tenant default_tenant. Are you sure it exists?

### ISSUE THREAD

**Issue by tusharvelingkar** [amikos-tech/chromadb-chart#57]
### What happened?

Getting default_tenant not found error when trying to connect to remote Chroma instance deployed on Kubernetes

Below are the logs which are seen in the pod

`Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
Traceback (most recent call last):  File "/chroma/venv/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/main.py", line 408, in main
    run(
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/main.py", line 576, in run
    server.run()
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/server.py", line 67, in serve
    config.load()
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/config.py", line 479, in load
    self.loaded_app = import_from_string(self.app)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/chroma/./chromadb/__init__.py", line 5, in <module>
    from chromadb.api import API
  File "/chroma/./chromadb/api/__init__.py", line 4, in <module>
    from chromadb.api.models.Collection import Collection
  File "/chroma/./chromadb/api/models/Collection.py", line 39, in <module>
    class Collection(BaseModel):
  File "/chroma/./chromadb/api/models/Collection.py", line 51, in Collection
    embedding_function: Optional[EmbeddingFunction] = ef.DefaultEmbeddingFunction(),
  File "/chroma/./chromadb/utils/embedding_functions.py", line 420, in DefaultEmbeddingFunction
    return ONNXMiniLM_L6_V2()
  File "/chroma/./chromadb/utils/embedding_functions.py", line 269, in __init__
    self.ort = importlib.import_module("onnxruntime")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/chroma/venv/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
    from onnxruntime.capi._pybind_state import ExecutionMode  # noqa: F401
  File "/chroma/venv/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 32, in <module>
    from .onnxruntime_pybind11_state import *  # noqa
AttributeError: _ARRAY_API not found
ImportError: numpy.core.multiarray failed to import
The above exception was the direct cause of the following exception:
SystemError: <built-in function __import__> returned a result with an exception set`

### ChromaDB Version

0.4.3

### Kubernetes Version

1.27.x

### Environment details

On-prem K8s

### Logs, Kubectl output, Manifests, etc.

```shell
Some module may need to rebuild instead e.g. with 'pybind11>=2.12'.
If you are a user of the module, the easiest solution will be to
downgrade to 'numpy<2' or try to upgrade the affected module.
We expect that some modules will need time to support NumPy 2.
Traceback (most recent call last):  File "/chroma/venv/bin/uvicorn", line 8, in <module>
    sys.exit(main())
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1157, in __call__
    return self.main(*args, **kwargs)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1078, in main
    rv = self.invoke(ctx)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 1434, in invoke
    return ctx.invoke(self.callback, **ctx.params)
  File "/chroma/venv/lib/python3.10/site-packages/click/core.py", line 783, in invoke
    return __callback(*args, **kwargs)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/main.py", line 408, in main
    run(
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/main.py", line 576, in run
    server.run()
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/server.py", line 60, in run
    return asyncio.run(self.serve(sockets=sockets))
  File "/usr/local/lib/python3.10/asyncio/runners.py", line 44, in run
    return loop.run_until_complete(main)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/server.py", line 67, in serve
    config.load()
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/config.py", line 479, in load
    self.loaded_app = import_from_string(self.app)
  File "/chroma/venv/lib/python3.10/site-packages/uvicorn/importer.py", line 21, in import_from_string
    module = importlib.import_module(module_str)
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/chroma/./chromadb/__init__.py", line 5, in <module>
    from chromadb.api import API
  File "/chroma/./chromadb/api/__init__.py", line 4, in <module>
    from chromadb.api.models.Collection import Collection
  File "/chroma/./chromadb/api/models/Collection.py", line 39, in <module>
    class Collection(BaseModel):
  File "/chroma/./chromadb/api/models/Collection.py", line 51, in Collection
    embedding_function: Optional[EmbeddingFunction] = ef.DefaultEmbeddingFunction(),
  File "/chroma/./chromadb/utils/embedding_functions.py", line 420, in DefaultEmbeddingFunction
    return ONNXMiniLM_L6_V2()
  File "/chroma/./chromadb/utils/embedding_functions.py", line 269, in __init__
    self.ort = importlib.import_module("onnxruntime")
  File "/usr/local/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "/chroma/venv/lib/python3.10/site-packages/onnxruntime/__init__.py", line 23, in <module>
    from onnxruntime.capi._pybind_state import ExecutionMode  # noqa: F401
  File "/chroma/venv/lib/python3.10/site-packages/onnxruntime/capi/_pybind_state.py", line 32, in <module>
    from .onnxruntime_pybind11_state import *  # noqa
AttributeError: _ARRAY_API not found
ImportError: numpy.core.multiarray failed to import
The above exception was the direct cause of the following exception:
SystemError: <built-in function __import__> returned a result with an exception set
```

**Comment by tazarov** [comment]
@tusharvelingkar Thanks for raising this. It is related to recent numpy 2.0 releases - https://github.com/chroma-core/chroma/issues/2418 and https://github.com/chroma-core/chroma/issues/2418

The specific issue you're facing is that older versions of Chroma image will try to reinstall `chromadb-hsnwlib` which in turn will require numpy>=1.x thus triggering a numpy 2.0 install that breaks this.

I'll do some testing to see if this can be addressed. In the meantime, try bumping your image version to Chroma 0.5.3 (use the official image as I haven't yet rebuilt the default one that ships with the chart)

**Comment by tazarov** [comment]
@tusharvelingkar, this should now be fixed in chart version 0.1.20. Note that older Chroma images < 0.4.24 will still exhibit the numpy 2.x problem so please make sure that you're using Chroma version >=0.4.24.

The new chart version uses 0.5.5 as default Chroma version

**Comment by tazarov** [comment]
closing as fixed
